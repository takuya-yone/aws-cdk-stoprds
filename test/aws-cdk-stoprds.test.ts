import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import {
  AwsCdkStopRdsSfnStack,
  AwsCdkStopRdsEventStack,
} from '../lib/aws-cdk-stoprds-stack';
import { Template } from 'aws-cdk-lib/assertions';

test('Unit Test', () => {
  const app = new cdk.App();
  const sfnStack = new AwsCdkStopRdsSfnStack(app, 'AwsCdkStopRdsSfnStack');
  const eventStack = new AwsCdkStopRdsEventStack(
    app,
    'AwsCdkStopRdsEventStack',
    {
      stopRdsStateMachine: sfnStack.stopRdsStateMachine,
    }
  );

  const template = Template.fromStack(sfnStack);
  expect(template.toJSON()).toMatchSnapshot();

  // template.hasResourceProperties("AWS::Lambda::Function", {
  //   Runtime: "nodejs18.x",
  // });
});
