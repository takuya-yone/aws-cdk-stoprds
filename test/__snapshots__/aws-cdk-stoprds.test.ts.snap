// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test 1`] = `
{
  "Mappings": {
    "ServiceprincipalMap": {
      "af-south-1": {
        "states": "states.af-south-1.amazonaws.com",
      },
      "ap-east-1": {
        "states": "states.ap-east-1.amazonaws.com",
      },
      "ap-northeast-1": {
        "states": "states.ap-northeast-1.amazonaws.com",
      },
      "ap-northeast-2": {
        "states": "states.ap-northeast-2.amazonaws.com",
      },
      "ap-northeast-3": {
        "states": "states.ap-northeast-3.amazonaws.com",
      },
      "ap-south-1": {
        "states": "states.ap-south-1.amazonaws.com",
      },
      "ap-south-2": {
        "states": "states.ap-south-2.amazonaws.com",
      },
      "ap-southeast-1": {
        "states": "states.ap-southeast-1.amazonaws.com",
      },
      "ap-southeast-2": {
        "states": "states.ap-southeast-2.amazonaws.com",
      },
      "ap-southeast-3": {
        "states": "states.ap-southeast-3.amazonaws.com",
      },
      "ca-central-1": {
        "states": "states.ca-central-1.amazonaws.com",
      },
      "cn-north-1": {
        "states": "states.cn-north-1.amazonaws.com",
      },
      "cn-northwest-1": {
        "states": "states.cn-northwest-1.amazonaws.com",
      },
      "eu-central-1": {
        "states": "states.eu-central-1.amazonaws.com",
      },
      "eu-central-2": {
        "states": "states.eu-central-2.amazonaws.com",
      },
      "eu-north-1": {
        "states": "states.eu-north-1.amazonaws.com",
      },
      "eu-south-1": {
        "states": "states.eu-south-1.amazonaws.com",
      },
      "eu-south-2": {
        "states": "states.eu-south-2.amazonaws.com",
      },
      "eu-west-1": {
        "states": "states.eu-west-1.amazonaws.com",
      },
      "eu-west-2": {
        "states": "states.eu-west-2.amazonaws.com",
      },
      "eu-west-3": {
        "states": "states.eu-west-3.amazonaws.com",
      },
      "me-central-1": {
        "states": "states.me-central-1.amazonaws.com",
      },
      "me-south-1": {
        "states": "states.me-south-1.amazonaws.com",
      },
      "sa-east-1": {
        "states": "states.sa-east-1.amazonaws.com",
      },
      "us-east-1": {
        "states": "states.us-east-1.amazonaws.com",
      },
      "us-east-2": {
        "states": "states.us-east-2.amazonaws.com",
      },
      "us-gov-east-1": {
        "states": "states.us-gov-east-1.amazonaws.com",
      },
      "us-gov-west-1": {
        "states": "states.us-gov-west-1.amazonaws.com",
      },
      "us-iso-east-1": {
        "states": "states.amazonaws.com",
      },
      "us-iso-west-1": {
        "states": "states.amazonaws.com",
      },
      "us-isob-east-1": {
        "states": "states.amazonaws.com",
      },
      "us-west-1": {
        "states": "states.us-west-1.amazonaws.com",
      },
      "us-west-2": {
        "states": "states.us-west-2.amazonaws.com",
      },
    },
  },
  "Outputs": {
    "ExportsOutputFnGetAttStopRdsStateMachineEventsRole04EB62A4Arn9C38159F": {
      "Export": {
        "Name": "AwsCdkStopRdsSfnStack:ExportsOutputFnGetAttStopRdsStateMachineEventsRole04EB62A4Arn9C38159F",
      },
      "Value": {
        "Fn::GetAtt": [
          "StopRdsStateMachineEventsRole04EB62A4",
          "Arn",
        ],
      },
    },
    "ExportsOutputRefStopRdsStateMachine82A35064CC83F663": {
      "Export": {
        "Name": "AwsCdkStopRdsSfnStack:ExportsOutputRefStopRdsStateMachine82A35064CC83F663",
      },
      "Value": {
        "Ref": "StopRdsStateMachine82A35064",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "StopRdsStateMachine82A35064": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "StopRdsStateMachineRoleDefaultPolicyF936C944",
        "StopRdsStateMachineRoleCAE3BD71",
      ],
      "Properties": {
        "DefinitionString": {
          "Fn::Join": [
            "",
            [
              "{"StartAt":"Wait 5min","States":{"Wait 5min":{"Type":"Wait","Seconds":300,"Next":"Stop Clusters"},"Stop Clusters":{"Type":"Parallel","End":true,"Branches":[{"StartAt":"StopRdsTask_0","States":{"StopRdsTask_0":{"End":true,"Retry":[{"ErrorEquals":["States.ALL"],"MaxAttempts":4,"BackoffRate":5}],"Type":"Task","Resource":"arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":states:::aws-sdk:rds:stopDBCluster","Parameters":{"DbClusterIdentifier":"nextjs-docker-aurora"}}}},{"StartAt":"StopRdsTask_1","States":{"StopRdsTask_1":{"End":true,"Retry":[{"ErrorEquals":["States.ALL"],"MaxAttempts":4,"BackoffRate":5}],"Type":"Task","Resource":"arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":states:::aws-sdk:rds:stopDBCluster","Parameters":{"DbClusterIdentifier":"portal-request-db-test-v2-cluster"}}}}]}}}",
            ],
          ],
        },
        "LoggingConfiguration": {
          "Destinations": [
            {
              "CloudWatchLogsLogGroup": {
                "LogGroupArn": {
                  "Fn::GetAtt": [
                    "stopRdsStateMachineLogGroup99160390",
                    "Arn",
                  ],
                },
              },
            },
          ],
          "Level": "ALL",
        },
        "RoleArn": {
          "Fn::GetAtt": [
            "StopRdsStateMachineRoleCAE3BD71",
            "Arn",
          ],
        },
        "StateMachineName": "StopRdsStateMachine",
        "TracingConfiguration": {
          "Enabled": true,
        },
      },
      "Type": "AWS::StepFunctions::StateMachine",
      "UpdateReplacePolicy": "Delete",
    },
    "StopRdsStateMachineEventsRole04EB62A4": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "StopRdsStateMachineEventsRoleDefaultPolicy5D70BB88": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "states:StartExecution",
              "Effect": "Allow",
              "Resource": {
                "Ref": "StopRdsStateMachine82A35064",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "StopRdsStateMachineEventsRoleDefaultPolicy5D70BB88",
        "Roles": [
          {
            "Ref": "StopRdsStateMachineEventsRole04EB62A4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "StopRdsStateMachineRoleCAE3BD71": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": {
                  "Fn::FindInMap": [
                    "ServiceprincipalMap",
                    {
                      "Ref": "AWS::Region",
                    },
                    "states",
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "StopRdsStateMachineRoleDefaultPolicyF936C944": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogDelivery",
                "logs:GetLogDelivery",
                "logs:UpdateLogDelivery",
                "logs:DeleteLogDelivery",
                "logs:ListLogDeliveries",
                "logs:PutResourcePolicy",
                "logs:DescribeResourcePolicies",
                "logs:DescribeLogGroups",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
                "xray:GetSamplingRules",
                "xray:GetSamplingTargets",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "rds:StopDBCluster",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:rds:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":cluster:nextjs-docker-aurora",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:rds:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":cluster:portal-request-db-test-v2-cluster",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "StopRdsStateMachineRoleDefaultPolicyF936C944",
        "Roles": [
          {
            "Ref": "StopRdsStateMachineRoleCAE3BD71",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "stopRdsStateMachineLogGroup99160390": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "RetentionInDays": 365,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
